src/components/Header/HeaderForm.tsx

```diff
@@ -0,0 +1,49 @@
+import { FC, useEffect, useRef, useState } from 'react';
+
+interface IProps {
+  addTodo: (newTodo: string) => void;
+}
+
+export const HeaderForm: FC<IProps> = ({ addTodo }) => {
+  const [newTodo, setNewTodo] = useState('');
+  const textInput = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    textInput.current?.focus();
+  }, []);
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    const trimmedTodo = newTodo.trim();
+
+    if (trimmedTodo !== '') {
+      addTodo(trimmedTodo);
+    }
+
+    setNewTodo('');
+  };
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTodo(e.target.value);
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <label htmlFor="NewTodo">
+        <input
+          id="NewTodo"
+          data-cy="NewTodoField"
+          type="text"
+          title="Write new todo"
+          className="todoapp__new-todo"
+          placeholder="What needs to be done?"
+          value={newTodo}
+          onChange={handleChange}
+          ref={textInput}
+        />
+      </label>
+      <button type="submit" style={{ display: 'none' }}></button>
```

Comment:
```suggestion
          title="Write new todo"
          className="todoapp__new-todo"
          placeholder="What needs to be done?"
          value={newTodo}
          onChange={handleChange}
          ref={textInput}
        />
      </label>
```
looks like we don't need this button at all